pkg/lockfile/parse-maven-lock.go

```diff
@@ -21,6 +24,57 @@ import (
 	"github.com/google/osv-scanner/internal/cachedregexp"
 )
 
+const MavenCentral = "https://repo.maven.apache.org/maven2"
+
+type MavenRegistryProject struct {
+	io.ReadCloser
+	path string
+}
+
+var errAPIFailed = errors.New("API query failed")
+
+func NewMavenRegistryAPIClient(url string) (*MavenRegistryProject, error) {
```

Comment:
I think it should return a DepFile for code readability (return the interface, not the implementation)

Reply:
And maybe it would be better in an internal package instead of the main lockfile one